File: iris-crypto/iris-crypto-rsa/src/main/java/city/smartb/iris/crypto/rsa/verifier/Verifier.java
package city.smartb.iris.crypto.rsa.verifier;

import com.nimbusds.jose.JWSAlgorithm;

import java.security.GeneralSecurityException;
import java.security.interfaces.RSAPublicKey;

public interface Verifier {

    static RS256Verifier rs256Verifier(RSAPublicKey rsaPublicKey) {
        return new RS256Verifier(rsaPublicKey);
    }

    JWSAlgorithm getAlgorithm();

    boolean verify(byte[] content, byte[] signature) throws GeneralSecurityException;

}

File: iris-crypto/iris-crypto-rsa-key/src/main/kotlin/city/smartb/iris/crypto/rsa/verifier/Verifier.kt
package city.smartb.iris.crypto.rsa.verifier

import com.nimbusds.jose.JWSAlgorithm
import java.security.GeneralSecurityException
import java.security.interfaces.RSAPublicKey

interface Verifier {
    val algorithm: JWSAlgorithm?

    @Throws(GeneralSecurityException::class)
    fun verify(content: ByteArray, signature: ByteArray): Boolean

    companion object {
        fun rs256Verifier(rsaPublicKey: RSAPublicKey): RS256Verifier {
            return RS256Verifier(rsaPublicKey)
        }
    }
}
